src/index.js

```diff
@@ -164,8 +240,30 @@ export default function createCopyOnWriteState<T>(baseState: T) {
       );
     };
 
+    getObservedState(state: T, selectors: Selector<T, S>): ObservedState<S> {
+      if (Array.isArray(selectors)) {
+        return selectors.map(selector => selector(state));
+      }
+      return selectors(state);
+    }
+
+    componentDidMount() {
```

Comment:
Note each componentDidMount/componentWillUnmount has a cost for React. Both running the code and including every fiber that has it in the effect list that is traversed during commit phase.

Reply:
Interesting, I didn't know that including these lifecycles had an inherent overhead. For the current implementation of the optimization strategy it needs to know when the selectors are being used, and when they stop being used.

I'm not sure of a way to do that without these commit phase lifecycles.

Reply:
Yeah, I dunno, just sayin'

Reply:
It's good to know either away, I appreciate it 👏